internal/authentication/tokengetter.go

```diff
@@ -86,7 +95,8 @@ func (t *TokenGetter) getToken(ctx context.Context, key types.NamespacedName) (*
 			Spec: authenticationv1.TokenRequestSpec{ExpirationSeconds: ptr.To(int64(t.expirationDuration / time.Second))},
 		}, metav1.CreateOptions{})
 	if err != nil {
-		return nil, err
+		saErr := &SANotFoundError{key.Name}
```

Comment:
Is it guaranteed that the only kind of error that is returned here is that the ServiceAccount we are requesting a token for is not found?

Reply:
@thetechnick is this a fair assumption, like discussed above since we are doing a GET for a serviceaccount that would be the case.

Reply:
I left another comment here with my thoughts on this: https://github.com/operator-framework/operator-controller/pull/1477#discussion_r1850476014